documentation/architecture.md

```diff
@@ -0,0 +1,337 @@
+# Clear Containers Architecture
+
+* [Overview](#overview)
+  * [Hypervisor](#hypervisor)
+  * [Hyperstart](#hyperstart)
+  * [Runtime](#runtime)
+    * [create](#create)
+    * [start](#start)
+    * [exec](#exec)
+    * [kill](#kill)
+    * [delete](#delete)
+  * [Proxy](#proxy)
+  * [Shim](#shim)
+  * [Networking](#networking)
+
+## Overview
+
+This is an architectural overview of Clear Containers, based on the 2.1 release.
+
+The [Clear Containers runtime (cc-oci-runtime)](https://github.com/01org/cc-oci-runtime)
+complies with the [OCI](https://github.com/opencontainers) specifications and thus
+works seamlessly with the [Docker Engine](https://www.docker.com/products/docker-engine)
+pluggable runtime architecture. In other words, one can transparently replace the
+[default Docker runtime (runc)](https://github.com/opencontainers/runc) with `cc-oci-runtime`.
+
+![Docker and Clear Containers](https://github.com/01org/cc-oci-runtime/blob/topic/documentation/documentation/docker-cc.png)
```

Comment:
Would it be possible to add the sources for these graphics files? That would make maintenance easier since without them, we'd probably have to regenerate the diagrams from scratch.